go/libraries/doltcore/merge/merge_rows.go

```diff
@@ -158,43 +179,73 @@ func (rm *RootMerger) MergeTable(ctx *sql.Context, tblName string, opts editor.O
 	return &MergedTable{table: tbl}, stats, nil
 }
 
-func (rm *RootMerger) makeTableMerger(ctx context.Context, tblName string) (*TableMerger, error) {
+func (rm *RootMerger) makeTableMerger(ctx context.Context, tblName string, mergeOpts MergeOpts) (*TableMerger, error) {
+	recordViolations := true
+	if mergeOpts.RecordViolationsForTables != nil {
+		if _, ok := mergeOpts.RecordViolationsForTables[strings.ToLower(tblName)]; !ok {
+			recordViolations = false
+		}
+	}
+
 	tm := TableMerger{
-		name:        tblName,
-		rightSrc:    rm.rightSrc,
-		ancestorSrc: rm.ancSrc,
-		vrw:         rm.vrw,
-		ns:          rm.ns,
+		name:             tblName,
+		rightSrc:         rm.rightSrc,
+		ancestorSrc:      rm.ancSrc,
+		vrw:              rm.vrw,
+		ns:               rm.ns,
+		recordViolations: recordViolations,
 	}
 
-	var ok bool
 	var err error
+	var leftSideTableExists, rightSideTableExists, ancTableExists bool
 
-	tm.leftTbl, ok, err = rm.left.GetTable(ctx, tblName)
+	tm.leftTbl, leftSideTableExists, err = rm.left.GetTable(ctx, tblName)
 	if err != nil {
 		return nil, err
 	}
-	if ok {
+	if leftSideTableExists {
 		if tm.leftSch, err = tm.leftTbl.GetSchema(ctx); err != nil {
 			return nil, err
 		}
 	}
 
-	tm.rightTbl, ok, err = rm.right.GetTable(ctx, tblName)
+	tm.rightTbl, rightSideTableExists, err = rm.right.GetTable(ctx, tblName)
 	if err != nil {
 		return nil, err
 	}
-	if ok {
+	if rightSideTableExists {
 		if tm.rightSch, err = tm.rightTbl.GetSchema(ctx); err != nil {
 			return nil, err
 		}
 	}
 
-	tm.ancTbl, ok, err = rm.anc.GetTable(ctx, tblName)
+	// If we need to re-verify all constraints, then we need to stub out tables
+	// that don't exist, so that the diff logic can compare an empty table to
+	// the table containing the real data. This is required by dolt_verify_constraints()
+	// so that we can run the merge logic on all rows in all tables.
+	if mergeOpts.ReverifyAllConstraints {
+		if !leftSideTableExists && rightSideTableExists {
```

Comment:
This is pretty convoluted and mixes a bunch of different concerns together. My primary objection is that we're altering the `tm` object in place here, which changes the semantics of what comes next for an unrelated concern. Seems like it would be better to special case just this logic and do a separate pass than try to cram this use case into the same merge code path. Or if that's not easy, then call the merge code with a copy of the table merger with these alterations.

Reply:
calling merge code with an edited copy sounds like a good near term idea

Reply:
maybe delete the tables whose constraints we don't want to verify? you wouldn't need `recordViolations` in that case also

Reply:
and then maybe `ReverifyAllConstraints` -> `shortCircuit = false`?

Reply:
I think it's valid to adjust `tm` in this function, since this is the constructor function for the `tm` instance and once created, this instance is stable for it's lifetime and doesn't get altered. I agree that the concerns are mixed together a bit though. 

Regarding deleting the tables with constraints we don't want to modify... I tried something similar first (skipping over the tables completely), but because the root gets written back in order to store the violation artifacts, this led to other problems in the new root. It did seem to me like this was going to work though, so maybe I just missed something in that attempt.